src/psyclone/configuration.py

```diff
@@ -748,27 +753,86 @@ class DynConfig(APISpecificConfig):
     # pylint: disable=too-few-public-methods
     def __init__(self, config, section):
         super(DynConfig, self).__init__(section)
-        self._config = config  # Ref. to parent Config object
-        try:
-            self._compute_annexed_dofs = section.getboolean(
-                'COMPUTE_ANNEXED_DOFS')
-        except ValueError as err:
-            raise ConfigurationError(
-                "error while parsing COMPUTE_ANNEXED_DOFS in the [dynamo0.3] "
-                "section of the config file: {0}".format(str(err)),
-                config=self._config)
+        # Ref. to parent Config object
+        self._config = config
+        # Initialise redundant computation setting
+        self._compute_annexed_dofs = None
+        # Initialise LFRic datatypes' default kinds (precisions) settings
+        self._default_kind = {}
+        # Set mandatory keys
+        # TODO: to be fully populated here for LFRic (Dynamo0.3) API in #282
+        # when Dynamo0.1 API is removed.
+        self._mandatory_keys = []
+
+        # TODO: This "if" clause will become redundant in #282 as there will be
+        # just LFRic (Dynamo0.3) configuration
+        if section.name == "dynamo0.3":
+            # Define and check mandatory keys
+            self._mandatory_keys = ["access_mapping",
```

Comment:
Keys in the config file are actually case-insensitive so we can have "compute_annexed_dofs" here and then you don't need the list comprehension in order to get lower-case names.

Reply:
Done. I left the upper case `COMPUTE_ANNEXED_DOFS` in `getboolean` error message for consistency with other boolean keys in the `[default]` section.